index.html

```diff
@@ -1595,6 +1573,17 @@ <h3>
         </dl>
       </section>
     </section>
+    <section id="same-origin-check">
+      <h3>Same-origin check</h3>
+      <p>
+        When asked to run the <dfn>same-origin check</dfn>, the user agent
+        must run the following: If the previous document, and all <a
```

Comment:
Should this have an input or specify 'current document' in a less hand-wavy way? Also, the previous document may not exist, and I don't think we covered all cases where it does not every time we call this, so maybe worth special casing that here too. Given that, we might also want to convert this to an algorithm.

Reply:
There are many underdefined concepts we use here and I don't want to block on them. The current document is used all over the document, previous document is not really defined, there's no link between the current document and the request on which it rode in, and HTTP redirects don't keep track if they were all same-origin or not...

I made it more algorithmy, but a lot of this stuff will have to wait for a larger scale Fetch integration.